### PR TITLE
feat: add support for plans with trial periods

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Add support for plans with trial periods (in days).

additive: support in data model, support in poke, pass the value (if any) to stripe when creating the checkout session.

Won't do anything until we actually have plans that use it.

## Risk

- migration

## Deploy Plan

front initdb then deploy front

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

File: front/components/poke/plans/form.tsx [hunk 1]
```diff
@@ -60,6 +61,7 @@ export const fromPlanType = (plan: PlanType): EditingPlanType => {
     dataSourcesDocumentsSizeMb: plan.limits.dataSources.documents.sizeMb,
     maxUsers: plan.limits.users.maxUsers,
     billingType: plan.billingType,
+    trialPeriodDays: plan.trialPeriodDays ?? 0,
```
Comment (spolu) [inline front/components/poke/plans/form.tsx:64]: So if we edit we move it from null to 0 right?

File: front/components/poke/plans/form.tsx [hunk 2]
```diff
@@ -97,6 +99,7 @@ export const toPlanType = (editingPlan: EditingPlanType): PlanType => {
       },
     },
     billingType: editingPlan.billingType,
+    trialPeriodDays: parseInt(editingPlan.trialPeriodDays.toString(), 10),
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/poke/plans/form.tsx:102]: What we do on every one of these "fields". The `toString()` isn't actually useful, will cleanup

File: front/components/poke/plans/form.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/poke/plans/form.tsx:64]: Just in the form. We still use `null` in DB

File: front/lib/plans/stripe.ts [hunk 3]
```diff
@@ -111,6 +111,7 @@ export const createCheckoutSession = async ({
         planCode: planCode,
         workspaceId: owner.sId,
       },
+      trial_period_days: plan.trialPeriodDays || undefined,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/plans/stripe.ts:114]: No, stripe wants either some number or undefined.

File: front/components/poke/plans/form.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/poke/plans/form.tsx:64]: Yes. Actually I can't really decide what is best ? Do we want to keep `null` or use 0 everywhere (and by default) ?

Comment (fontanierh) [reply front/components/poke/plans/form.tsx:64]: I feel like there's not really a reason to use nulls. I'll just go `0` everywhere (and have non-nullable, 0 as default in schema)

Comment (spolu) [reply front/components/poke/plans/form.tsx:64]: I'm fine with either or but I think we want consistency. Aka all null if non zero or all zero :+1:

Comment (fontanierh) [reply front/components/poke/plans/form.tsx]: We don't send the `0` to stripe. We replace by `undefined`. We want to have this logic before sending to strip regardless of nullability, so IMO having 0 is just better (consistency is guaranteed, less ambiguous)

**Pull request merged**